### PR TITLE
Added option -min-level to filter issues by severity

### PR DESCRIPTION
**Pull request by sergey-shambir**
Also did refactoring:
- moved flags to flags.go (but global variables are evil)
- added new type SeverityLevel same as int
Syntax:
```bash
 -min-level=error|warning|info|hint|unused|maybe|syntax
```
Relates to issue #66

### CODE REVIEW

**Comment by quasilyte** [comment]
I'll take a look today, thank.

**Comment by quasilyte** [comment]
It would be easier to review if refactoring happened separately from the feature implementation.

File: src/linter/custom.go [hunk 1]
```diff
@@ -176,14 +179,14 @@ var vscodeLevelMap = map[int]int{
...
-	LevelDoNotReject: "MAYBE  ",
-	LevelSyntax:      "SYNTAX ",
+var severityLevelNames = map[SeverityLevel]string{
+	LevelError:       "error",
+	LevelWarning:     "warning",
+	LevelInformation: "info",
+	LevelHint:        "hint",
+	LevelUnused:      "unused",
+	LevelDoNotReject: "maybe",
+	LevelSyntax:      "syntax",
```
Comment (quasilyte) [inline src/linter/custom.go]: I'm not sure about this severity level distribution. If you use min level filtering, you need a good sorting here and I don't know how to rank them from 1 to 7 properly. Are you sure you don't want to filter by using string list of prefixes?

Comment (sergey-shambir) [reply src/linter/custom.go]: I'm agree, it's better to pass enabled severity list.
However, it's inconvient to pass levels with command line flags.
So I suggest:
1) add config file support with separate PR
2) after that, add filtering by severity level names

**Comment by quasilyte** [comment]
We should discuss and design the warning level/configuration separately.
I've applied your refactoring in #95.

**Comment by quasilyte** [comment]
Discussion can be continued in #83.
Thanks.

**Pull request closed without merging**